### ISSUE TITLE
Feature Idea: Speed Control Instead Of Time-Keyframes

### ISSUE THREAD

**Issue by Kepler-17c** [ReplayMod/ReplayMod#721]
The idea arose from a discussion in general chat: https://discord.com/channels/708304818829262910/708317114183843850/965025973504471060

TL;DR: Using time-keyframes can be confusing and has a steep learning curve. Having control over the replay-speed is more intuitive than placing time-keyframes.

This feature would add a different keyframing mode to ReplayMod with the following properties:  
* position-keyframes are marked on the game-timeline
  * the goal is to create a video-like base
* the video-timeline works similar to video editing:  
  * set "cuts" to mark sections
  * sections have a label with the speed factor
  * stretching or compressing sections changes the speed and moves surrounding sections
  * clicking on the label enables editing the factor and stretches/compresses the section accordingly
* video-editor-like cutting could be based on the existing queue functionality
  * a normal "cut" just places a marker
  * all connected segments (in a sense of game-time) make up one path
  * removing a section splits the path in two and will queue them for rendering

How it would translate to the existing timeline format:  
* each section split still secretly is a time-keyframe
* the speed factor is calculated from the two surrounding keyframes and their time values
* stretching/compressing a section moves the time-keyframe and all following ones in video-time
* position-keyframes require an intermediate that stores the game-time and translates it to video-time using the time-keyframes
